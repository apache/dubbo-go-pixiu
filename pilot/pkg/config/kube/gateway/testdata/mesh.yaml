apiVersion: gateway.networking.k8s.io/v1alpha2
kind: GatewayClass
metadata:
  name: istio
spec:
  controllerName: istio.io/gateway-controller
---
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: Gateway
metadata:
  name: gateway
  namespace: istio-system
spec:
  addresses:
  - value: istio-ingressgateway
    type: Hostname
  gatewayClassName: istio
  listeners:
  - name: default
    hostname: "*.example.com"
    port: 80
    protocol: HTTP
    allowedRoutes:
      namespaces:
        from: All
---
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: HTTPRoute
metadata:
  name: echo
  namespace: default
spec:
  parentRefs:
  - kind: Mesh
    name: istio
  hostnames: ["echo.default.svc.cluster.local"]
  rules:
  - backendRefs:
    - name: echo
      port: 80
---
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: TLSRoute
metadata:
  name: echo
  namespace: default
spec:
  parentRefs:
  - kind: Mesh
    name: istio
  hostnames: ["some-sni.com"]
  rules:
  - backendRefs:
    - name: echo
      port: 80
---
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: HTTPRoute
metadata:
  name: dual # applies to mesh and explicit gateway
  namespace: default
spec:
  parentRefs:
  - kind: Mesh
    name: istio
  - name: gateway
    namespace: istio-system
  hostnames: ["foo.example.com"]
  rules:
  - backendRefs:
    - name: example
      port: 80
---
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: HTTPRoute
metadata:
  name: no-hostname
  namespace: default
spec:
  parentRefs:
  - kind: Mesh
    name: istio
  rules:
  - backendRefs:
    - name: example
      port: 80
