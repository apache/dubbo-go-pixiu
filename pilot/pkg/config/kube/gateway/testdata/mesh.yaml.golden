apiVersion: networking.istio.io/v1alpha3
kind: Gateway
metadata:
  annotations:
    internal.istio.io/gateway-service: istio-ingressgateway.istio-system.svc.domain.suffix
    internal.istio.io/parent: Gateway/gateway/default.istio-system
  creationTimestamp: null
  name: gateway-istio-autogenerated-k8s-gateway-default
  namespace: istio-system
spec:
  servers:
  - hosts:
    - '*/*.example.com'
    port:
      name: default
      number: 80
      protocol: HTTP
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parent: HTTPRoute/echo.default
    internal.istio.io/route-semantics: gateway
  creationTimestamp: null
  name: echo-0-istio-autogenerated-k8s-gateway
  namespace: default
spec:
  gateways:
  - mesh
  hosts:
  - echo.default.svc.cluster.local
  http:
  - route:
    - destination:
        host: echo.default.svc.domain.suffix
        port:
          number: 80
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parent: HTTPRoute/dual.default
    internal.istio.io/route-semantics: gateway
  creationTimestamp: null
  name: dual-0-istio-autogenerated-k8s-gateway
  namespace: default
spec:
  gateways:
  - istio-system/gateway-istio-autogenerated-k8s-gateway-default
  hosts:
  - foo.example.com
  http:
  - route:
    - destination:
        host: example.default.svc.domain.suffix
        port:
          number: 80
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parent: HTTPRoute/dual.default
    internal.istio.io/route-semantics: gateway
  creationTimestamp: null
  name: dual-1-istio-autogenerated-k8s-gateway
  namespace: default
spec:
  gateways:
  - mesh
  hosts:
  - foo.example.com
  http:
  - route:
    - destination:
        host: example.default.svc.domain.suffix
        port:
          number: 80
---
